File: backend/src/main/java/com/ll/hotel/domain/hotel/option/entity/BaseOption.java
package com.ll.hotel.domain.hotel.option.entity;

import com.ll.hotel.global.jpa.entity.BaseEntity;
import jakarta.persistence.Column;
import jakarta.persistence.MappedSuperclass;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.*;

@Getter
@Setter
@MappedSuperclass
@NoArgsConstructor
@EqualsAndHashCode(onlyExplicitlyIncluded = true, callSuper = false)
public abstract class BaseOption extends BaseEntity {
    @NotBlank(message = "필수 항목입니다.")
    @Size(max = 255, message = "최대 255자까지 작성 가능합니다.")
    @EqualsAndHashCode.Include
    @Column(unique = true, nullable = false, length = 255)
    protected String name;

    public BaseOption(String name) {
        this.name = name;
    }
}

File: backend/src/main/java/com/ll/hotel/domain/hotel/option/entity/BaseOption.kt
package com.ll.hotel.domain.hotel.option.entity

import com.ll.hotel.global.jpa.entity.BaseEntity
import jakarta.persistence.MappedSuperclass
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

@MappedSuperclass
abstract class BaseOption(
    @field:NotBlank
    @field:Size(max = 255)
    var name: String
): BaseEntity() {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is BaseOption) return false
        if (name != other.name) return false
        return true
    }
    override fun hashCode(): Int {
        return name.hashCode()
    }
}
